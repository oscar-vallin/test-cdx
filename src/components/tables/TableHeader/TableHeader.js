import React from 'react';
import PropTypes from 'prop-types';
import { DetailsHeader } from 'office-ui-fabric-react/lib/DetailsList';
import { useHistory } from 'react-router-dom';

import { HeaderTable, StyledColumn, StyledRow, StyledMenuButton, StyledText, StyledLink } from './TableHeader.styles';

const TableHeader = ({ id = '__TableHeader', header = 'default', sortLabel, onSort, onOption, ...props }) => {
  const history = useHistory();

  console.log({ header });

  if ((!header || !header.type || header.type === 'default') && props) {
    return <DetailsHeader {...props} ariaLabelForToggleAllGroupsButton="Toggle selection" />;
  }

  if (header.type === 'dashboard') {
    return (
      <HeaderTable>
        <StyledColumn left paddingLeft={12}>
          <StyledLink onClick={() => history.push('/transmissions')}>{header.title}</StyledLink>
        </StyledColumn>
        {header.buttons && (
          <StyledColumn>
            <StyledRow>
              <StyledColumn>
                <StyledMenuButton
                  icon="sort"
                  onClick={() => {
                    onSort();
                  }}
                >
                  {sortLabel}
                </StyledMenuButton>
              </StyledColumn>
              <StyledColumn>
                <StyledMenuButton
                  icon="eye"
                  onClick={() => {
                    onOption();
                  }}
                >
                  {header.buttons[1]}
                </StyledMenuButton>
              </StyledColumn>
            </StyledRow>
          </StyledColumn>
        )}
      </HeaderTable>
    );
  }
  return null;
};

TableHeader.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
};

export { TableHeader };
