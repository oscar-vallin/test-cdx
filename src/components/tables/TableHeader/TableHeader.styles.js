import styled from 'styled-components';
import { Box as LayoutBox, Row as LayoutRow, Column as LayoutColumn } from '../../layouts';
import { Text } from '../../typography/Text';
import { Link } from '../../buttons/Link';
import { ButtonAction } from '../../buttons/ButtonAction';

export const StyledBox = styled(LayoutBox)`
  width: 100%;
`;

export const StyledRow = styled(LayoutRow)`
  width: 100%;
  margin-left: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  margin-right: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  border-bottom: gray;
  border-bottom-width: ${({ noBorder }) => (noBorder ? '0px' : '2px')};
  border-bottom-style: solid;
`;

export const StyledColumn = styled(LayoutColumn)`
  padding-left: ${({ paddingLeft }) => (paddingLeft ? `${paddingLeft}px` : '0px')};
  margin-left: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  margin-right: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
`;

export const StyledColumnTitle = styled(StyledColumn)`
  height: 100%;
`;

export const HeaderTable = styled(StyledRow)`
  height: 5vh;
  min-height: 38px;
`;

export const StyledText = styled(Text)`
  display: flex;
  width: 100%;
  font-weight: ${({ bold, theme }) => (bold ? theme.fontWeights.bold : theme.fontWeights.normal)};
  justify-content: ${({ left, right }) => (left ? 'flex-start' : right ? 'flex-end' : 'center')};
  align-items: center;
`;

export const StyledMenuButton = styled(ButtonAction)`
  color: ${({ theme }) => theme.colors.black};
`;

export const StyledLink = styled(Link)`
  width: 100%;
`;

export const RouteLink = styled(Link)`
  text-decoration: inherit;
  color: inherit;
`;
