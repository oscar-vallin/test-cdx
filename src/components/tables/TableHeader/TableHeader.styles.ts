import styled from 'styled-components';
import { Box as LayoutBox, Row as LayoutRow, Column as LayoutColumn } from '../../layouts';
import { Text } from '../../typography/Text';
import { Link } from '../../buttons/Link';
import { ButtonAction } from '../../buttons/ButtonAction';

export const StyledBox = styled(LayoutBox)`
  width: 100%;
`;

type RowProps = {
  marginH?: boolean;
  noBorder?: boolean;
};

export const StyledRow = styled(LayoutRow)<RowProps>`
  width: 100%;
  margin-left: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  margin-right: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  border-bottom: gray;
  border-bottom-width: ${({ noBorder }) => (noBorder ? '0px' : '2px')};
  border-bottom-style: solid;
`;

type ColumnProps = {
  marginH?: boolean;
  paddingLeft?: boolean;
};

export const StyledColumn = styled(LayoutColumn)<ColumnProps>`
  padding-left: ${({ paddingLeft }) => (paddingLeft ? `${paddingLeft}px` : '0px')};
  margin-left: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
  margin-right: ${({ marginH }) => (marginH ? `${marginH}px` : '0px')};
`;

export const StyledColumnTitle = styled(StyledColumn)`
  height: 100%;
`;

export const HeaderTable = styled(StyledRow)`
  height: 5vh;
  min-height: 38px;
`;

type TextProps = {
  left?: boolean;
};

export const StyledText = styled(Text)<TextProps>`
  display: flex;
  width: 100%;
  font-weight: ${({ bold, theme }) => (bold ? theme.fontWeights.bold : theme.fontWeights.normal)};
  justify-content: ${({ left, right }) => {
    if (left) return 'flex-start';

    if (right) return 'flex-end';

    return 'center';
  }};
  align-items: center;
`;

export const StyledMenuButton = styled(ButtonAction)`
  color: ${({ theme }) => theme.colors.black};
`;

export const StyledLink = styled(Link)`
  width: 100%;
`;

export const RouteLink = styled(Link)`
  text-decoration: inherit;
  color: inherit;
`;
