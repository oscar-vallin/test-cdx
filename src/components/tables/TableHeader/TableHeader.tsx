import PropTypes from 'prop-types';
import { DetailsHeader, DetailsListLayoutMode } from '@fluentui/react';
import { useHistory } from 'react-router-dom';
import { getDates } from 'src/helpers/tableHelpers.service';

import { useOrgSid } from 'src/hooks/useOrgSid';
import {
  HeaderTable,
  StyledColumn,
  StyledColumnTitle,
  StyledLink,
  StyledMenuButton,
  StyledRow,
} from './TableHeader.styles';
import { yyyyMMdd } from 'src/utils/CDXUtils';

const TableHeader = ({ id, header, sortLabel, onSort, onOption, date, ...props }) => {
  const { orgSid } = useOrgSid();
  const history = useHistory();
  if ((!header || !header.type || header.type === 'default') && props) {
    return (
      <DetailsHeader
        layoutMode={DetailsListLayoutMode.justified}
        {...props}
        ariaLabelForToggleAllGroupsButton="Toggle selection"
      />
    );
  }

  const startFormatted = yyyyMMdd(getDates(date).startDate);
  const endFormatted = yyyyMMdd(getDates(date).endDate);

  if (header.type === 'dashboard') {
    return (
      <HeaderTable id={`${id}-HeaderTable_dashboard`}>
        <StyledColumnTitle center paddingLeft sm={6}>
          <StyledLink
            onClick={() =>
              history.push(`/transmissions?startDate=${startFormatted}&endDate=${endFormatted}&orgSid=${orgSid}`)
            }
          >
            {header.title}
          </StyledLink>
        </StyledColumnTitle>
        {header.buttons && (
          <StyledColumn sm={6}>
            <StyledRow noBorder>
              <StyledColumn sm={6}>
                <StyledMenuButton
                  id="__SortButton"
                  icon="sort"
                  onClick={() => {
                    onSort();
                    return null;
                  }}
                >
                  {sortLabel}
                </StyledMenuButton>
              </StyledColumn>
              <StyledColumn sm={6}>
                <StyledMenuButton
                  id="__EyeButton"
                  icon="eye"
                  onClick={() => {
                    onOption();
                    return null;
                  }}
                >
                  {header.buttons[1]}
                </StyledMenuButton>
              </StyledColumn>
            </StyledRow>
          </StyledColumn>
        )}
      </HeaderTable>
    );
  }
  return null;
};

TableHeader.propTypes = {
  id: PropTypes.string,
  type: PropTypes.string,
};

export { TableHeader };
