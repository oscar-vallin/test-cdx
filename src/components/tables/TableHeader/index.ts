import { TableHeader } from 'src/components/tables/TableHeader/TableHeader';

export { TableHeader };
