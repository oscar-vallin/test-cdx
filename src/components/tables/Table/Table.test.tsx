import { mount, shallow, render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { DetailsList, DetailsListLayoutMode, SelectionMode } from 'office-ui-fabric-react/lib-commonjs/DetailsList';
import { Table as Component } from './index.js';
import { TableHeader } from '../TableHeader';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/file-status?orgSid=1',
  }),
}));

const defaultProps = {
  items: [
    {
      child: undefined,
      columnId: 'datetime',
      id: 'datetime',
      sublabel: undefined,
      text: '11/04/2020 08:12 AM',
      value: '11/04/2020 08:12 AM',
    },
    { child: undefined, columnId: 'planSponsor', id: 'planSponsor', sublabel: undefined, text: 'PZZA', value: 'PZZA' },
    { child: undefined, columnId: 'vendorId', id: 'vendorId', sublabel: undefined, text: 'Disc', value: 'Disc' },
    { child: undefined, columnId: 'specId', id: 'specId', sublabel: undefined, text: 'DiscSpec', value: 'DiscSpec' },
    {
      child: undefined,
      columnId: 'implementation',
      id: 'implementation',
      sublabel: undefined,
      text: 'DiscImplementation',
      value: 'DiscImplementation',
    },
    {
      child: undefined,
      columnId: 'inboundFilename',
      id: 'inboundFilename',
      sublabel: undefined,
      text: 'K2UDEMO-MetLife-TEST-Warnings.xml',
      value: 'K2UDEMO-MetLife-TEST-Warnings.xml',
    },
    {
      child: undefined,
      columnId: 'outboundFilename',
      id: 'outboundFilename',
      sublabel: undefined,
      text: 'K2UDEMO-MetLife-12282020085407.txt',
      value: 'K2UDEMO-MetLife-12282020085407.txt',
    },
    {
      child: undefined,
      columnId: 'outboundFilesize',
      id: 'outboundFilesize',
      sublabel: undefined,
      text: 45919,
      value: 45919,
    },
    {
      child: undefined,
      columnId: 'billingCount',
      id: 'billingCount',
      sublabel: undefined,
      text: 89,
      value: 89,
    },
    {
      child: undefined,
      columnId: 'totalRecords',
      id: 'totalRecords',
      sublabel: undefined,
      text: 132,
      value: 132,
    },
    {
      child: undefined,
      columnId: 'extractType',
      id: 'extractType',
      sublabel: undefined,
      text: 'Enrollment',
      value: 'Enrollment',
    },
    {
      child: undefined,
      columnId: 'extractVersion',
      id: 'extractVersion',
      sublabel: undefined,
      text: '1.0.0.2020092018',
      value: '1.0.0.2020092018',
    },
  ],
  columns: [
    { key: 'datetime', label: 'Delivered On', style: 'text' },
    { key: 'planSponsor', label: 'Plan Sponsor', style: 'text' },
    { key: 'vendorId', label: 'Vendor', style: 'text' },
    { key: 'specId', label: 'Spec', style: 'text' },
    { key: 'implementation', label: 'Implementation', style: 'text' },
    { key: 'inboundFilename', label: 'Client File', style: 'link' },
    { key: 'outboundFilename', label: 'Vendor File', style: 'link' },
    { key: 'outboundFilesize', label: 'Outbound File Size', style: 'text' },
    { key: 'billingCount', label: 'Billing Unit Count', style: 'text' },
    { key: 'totalRecords', label: 'Total Records', style: 'text' },
    { key: 'extractType', label: 'Feed', style: 'text' },
    { key: 'extractVersion', label: 'Version', style: 'text' },
  ],
  structure: {
    header: {
      type: 'dashboard',
    },
  },
  onOption: () => null,
  groups: [],
  searchInput: '',
  date: 'today',
  onItemsListChange: () => null,
  loading: false,
};

const defaultHeaderProps = {
  id: '__TableHeader',
  header: {
    type: 'dashboard',
    buttons: true,
  },
  sortLabel: 'sortLabel',
  onSort: () => null,
  onOption: () => null,
  date: 'today',
};

test('Matches Snapshot', () => {
  const wrapper = shallow(<Component {...defaultProps} />);

  expect(toJSON(wrapper)).toMatchSnapshot();
});

describe('Basic Table Component', () => {
  it('Should be defined', () => {
    expect(Component).toBeDefined();
  });

  it('Should render table header when structure is type dashboard', () => {
    const tree = shallow(<TableHeader {...defaultHeaderProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('Should render table header when structure is type dashboard', () => {
    const tree = shallow(<TableHeader {...defaultHeaderProps} />);
    expect(tree.prop('id')).toEqual('__TableHeader-HeaderTable_dashboard');
  });

  it('Should render table header correctly', () => {
    const tree = shallow(
      <Component
        {...defaultProps}
        structure={{
          header: {
            type: 'other',
          },
        }}
      />
    );
    expect(tree.prop('id')).toEqual('Table_Detailed');
  });

  it('Should render dashboard list item with props', () => {
    const tree = shallow(<Component {...defaultProps} />);
    const list = tree.find(DetailsList);

    expect(list.prop('className')).toEqual('root-109');
    expect(list.prop('id')).toEqual('TableDetailedList');
    expect(list.prop('selectionMode')).toEqual(SelectionMode.none);
    expect(list.prop('layoutMode')).toEqual(DetailsListLayoutMode.justified);
    expect(list.prop('setKey')).toEqual('none');
    expect(list.prop('isHeaderVisible')).toBeTruthy();
    expect(typeof list.prop('onRenderDetailsHeader')).toBe('function');
    expect(typeof list.prop('onRenderItemColumn')).toBe('function');
  });

  it('Should render no dashboard list item with props', () => {
    const tree = shallow(
      <Component
        {...defaultProps}
        structure={{
          header: {
            type: 'other',
          },
        }}
      />
    );
    const list = tree.find(DetailsList);
    expect(list.prop('className')).toEqual('root-109');
    expect(list.prop('id')).toEqual('TableDetailedList');
    expect(list.prop('selectionMode')).toEqual(SelectionMode.none);
    expect(list.prop('layoutMode')).toEqual(DetailsListLayoutMode.justified);
    expect(list.prop('setKey')).toEqual('none');
    expect(list.prop('isHeaderVisible')).toBeTruthy();
    expect(list.prop('onRenderDetailsHeader')).toEqual(null);
    expect(typeof list.prop('onRenderDetailsHeader')).not.toBe('function');
    expect(typeof list.prop('onRenderItemColumn')).toBe('function');
  });

  it('Should has items with props', () => {
    const tree = shallow(<Component {...defaultProps} />);
    const list = tree.find(DetailsList);
    expect(list.prop('items')).toEqual([]);
  });
});
