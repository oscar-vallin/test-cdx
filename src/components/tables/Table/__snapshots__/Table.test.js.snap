// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Testing Unit... Should render correctly 1`] = `
<styled.div
  id="Table_Detailed"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <StyledWithViewportComponent
    className="root-40"
    columns={
      Array [
        Object {
          "ariaLabel": "Operations for vendor",
          "columnActionsMode": 1,
          "fieldName": "vendor",
          "isCollapsable": false,
          "isCollapsible": false,
          "isGrouped": false,
          "isMultiline": false,
          "isResizable": undefined,
          "isRowHeader": false,
          "isSorted": false,
          "isSortedDescending": false,
          "key": "vendor",
          "maxWidth": 300,
          "minWidth": 100,
          "name": "vendor",
          "onColumnClick": undefined,
          "onColumnContextMenu": undefined,
        },
        Object {
          "ariaLabel": "Operations for bus",
          "columnActionsMode": 1,
          "fieldName": "bus",
          "isCollapsable": true,
          "isCollapsible": true,
          "isGrouped": false,
          "isMultiline": false,
          "isResizable": undefined,
          "isRowHeader": false,
          "isSorted": false,
          "isSortedDescending": false,
          "key": "bus",
          "maxWidth": 300,
          "minWidth": 100,
          "name": "bus",
          "onColumnClick": undefined,
          "onColumnContextMenu": undefined,
        },
      ]
    }
    id="TableDetailedList"
    isHeaderVisible={true}
    items={
      Array [
        Object {
          "bus": "10",
          "vendor": "Record 1",
        },
        Object {
          "bus": "20",
          "vendor": "Record 2",
        },
        Object {
          "bus": "30",
          "vendor": "Record 3",
        },
      ]
    }
    layoutMode={1}
    onItemInvoked={[Function]}
    onRenderDetailsHeader={[Function]}
    onRenderItemColumn={[Function]}
    selectionMode={0}
    setKey="none"
  />
</styled.div>
`;
